services/wireguard/resources/allocator.go

```diff
@@ -17,29 +17,155 @@
 
 package resources
 
-import "net"
+import (
+	"errors"
+	"fmt"
+	"net"
+	"strconv"
+	"strings"
+	"sync"
+)
+
+const interfacePrefix = "myst"
 
 // Allocator is mock wireguard resource handler.
 // It will manage lists of network interfaces names, IP addresses and port for endpoints.
-type Allocator struct{}
+type Allocator struct {
+	Ifaces      map[int]struct{}
+	IPAddresses map[int]struct{}
+	Ports       map[int]struct{}
+	mu          sync.Mutex
+}
+
+// NewAllocator creates new resource pool for wireguard connection.
+func NewAllocator() Allocator {
+	return Allocator{
+		Ifaces:      make(map[int]struct{}),
+		IPAddresses: make(map[int]struct{}),
+		Ports:       make(map[int]struct{}),
+	}
+}
+
+// AbandonedInterfaces returns a list of abandoned interfaces that exist in the system,
+// but was not allocated by the Allocator.
+func (a *Allocator) AbandonedInterfaces() ([]net.Interface, error) {
+	a.mu.Lock()
+	defer a.mu.Unlock()
+
+	ifaces, err := net.Interfaces()
+	if err != nil {
+		return nil, err
+	}
+
+	list := make([]net.Interface, 0)
+	for _, iface := range ifaces {
+		if strings.HasPrefix(iface.Name, interfacePrefix) {
+			ifaceID, err := strconv.Atoi(strings.TrimPrefix(iface.Name, interfacePrefix))
+			if err == nil {
```

Comment:
might want to log in case of error, for instance if we start creating some phantom interfaces we won't even know about it

Reply:
Added logging to the place with actual cleaning, not just listing.